doc/man/man5/strisdn.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH @PACKAGE_UCNAME@ 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System File Formats"
.SH NAME
.B @PACKAGE@
\- @PACKAGE_TITLE@ package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@-@VERSION@ )
package for
.BR "Linux Fast-STREAMS" .
.[
lfs
.]
.PP
.BR "Linux Fast-STREAMS" ,
.BR streams (5),
consists of a base package that provides basic
.I SVR 4.2 STREAMS
capabilities for
.B Linux
and a number of separate add-on packages.
.PP
OpenSS7
.I STREAMS
add-on packages provide additional functionality and protocol support not traditionally found on
UNIX systems, or which have traditionally fallen into the domain of value-add system software, or,
in this case, where functionality was optional or differently handled in a number of
implementations.
.PP
The
.B @PACKAGE_TITLE@
package is one of the several optional add-on packages that provide support for ISDN and is
part of the thrust of the
.IR "OpenSS7 Project" .
.[
%T The OpenSS7 Project
%O http://www.openss7.org
.]
.\"
.\"
.SH DESCRIPTION
.PP
The
.B @PACKAGE_TITLE@
package provides
.I STREAMS
modules and drivers, libraries, programs, init scripts, and daemons that
implement the
.I STREAMS
ISDN subsystem.
.PP
The package builds and installs the following
.I STREAMS
modules and drivers (as kernel modules):
.IP \(en \w'\(en\(em'u
.B streams_lapd.ko
driver
.IR /dev/lapd .
.PD 0
.IP \(en
.B streams_isdn.ko
driver
.IR /dev/isdn .
.PD
.PP
The following are
.I STREAMS
modules installed by the
.B @PACKAGE@
package:
.TP
.BR lapd (4)
The
.BR lapd (4)
driver is a multiplexing pseudo-device driver that links
.BR hdlc (4)
Streams beneath it.  It uses the Communications Device Interface (CDI),
.BR cdi (7),
at its lower multiplex boundary and the Data Link Provider Interface (DLPI),
.[
dlpi revision 2.0.0
.]
.BR dlpi (7),
at its upper multiplex boundary.
.TP
.BR isdn (4)
The
.BR isdn (4)
driver is a multiplexing pseudo-device driver that links
.BR lapd (4)
or
.BR iua (4)
Streams beneath it.  It uses the Data Link Provider Interface (DLPI),
.[
dlpi revision 2.0.0
.]
.BR dlpi (7),
at its lower multiplex boundary and the Call Control Interface (CCI),
.BR cci (7),
at its upper multiplex boundary.
.TP
.BR capi (4)
The
.BR capi (4)
driver is an implementation of the Common ISDN API (CAPI), Version 2.0.
.[
capi20
.]
This driver provides
.I STREAMS
access
(both
.IR STREAMS -based
CAPI as well as the
Call Control Interface (CCI),
.BR cci (7))
to
.B Linux
CAPI devices as well as providing Linux CAPI access to
.BR isdn (4),
.BR dass (4)
and
.BR dpnss (4)
devices.
.IP ""
This driver is currently incomplete.
.TP
.BR dass (4)
The
.BR dass (4)
driver is a DASS
.[
dass2
.]
version of the
.BR isdn (4)
driver.
.IP ""
This driver is currently incomplete.
.TP
.BR dpnss (4)
The
.BR dpnss (4)
driver is a DPNSS
.[
dpnss1
.]
version of the
.BR isdn (4)
driver.
.IP ""
This driver is currently incomplete.
.TP
.BR gr303 (4)
.IP ""
This driver is currently incomplete.
.TP
.BR v52 (4)
.IP ""
This driver is currently incomplete.
.TP
.BR idlc (4)
.IP ""
This driver is currently incomplete.
.TP
.BR lapf (4)
.PP
The
.B @PACKAGE@
package also provides the following service interface definitions:
.TP
.BR cci (7)
The
Call Control Interface (CCI),
.BR cci (7),
is a call control interface for use by ISDN, ISUP, H.323, SIPT and BICC,
developed by the OpenSS7 Project.  Because
.BR strisdn (5)
is the first add-on package in the dependency chain to require the Call
Control Interface, it also provides the interface.  This interface is also
used by the
.BR strss7 (5),
.BR sigtran (5)
and
.BR strvoip (5)
add-on packages.
.PP
The
.B @PACKAGE_TITLE@
package includes kernel modules,
.I SVR 4.2 STREAMS
drivers, modules, libraries, utilities, test programs, daemons, and development environment for the
development and execution of
.B @PACKAGE_TITLE@
protocol stack components and applications for the
.I SVR 4.2 STREAMS
environment.
.PP
This distribution is only currently applicable to
.B Linux
2.4 and 2.6 kernels and was targeted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
architectures, but should build and install for other architectures as well.
.PP
This package represents ISDN components that were previously part of the
.BR strss7 (5)
package, but which were separated into this package at the end of 2006.  Although this manual page
documents all commercial components in the
.BR strisdn (5)
package, not all components have been released publicly.
.\"
.\"
.SH "PACKAGE CONTENTS"
.PP
The
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@ - @VERSION@ )
package contains the following manual pages:
.HP
.BR @PACKAGE@ (5)\ \-\ 
.B @PACKAGE_TITLE@
package.
.PD 0
.HP
.BR @PACKAGE@_mknod (8)\ \-\ 
create or remove
.I STREAMS
VOIP device nodes.
.PD
.\"
.PP
.HP
.BR capi (4)\ \-\ 
Common ISDN API (CAPI) Device Driver.
.PD 0
.HP
.BR capi_ioctl (4)\ \-\ 
COMMON-ISDN-API (CAPI) - IO Controls.
.HP
.BR dass (4)\ \-\ 
OpenSS7 ISDN Devices.
.HP
.BR dl_gr303 (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR dl_lapd (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR dl_lapf (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR dl_v52 (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR dpnss (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR gr303 (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR gr303_ioctl (4)\ \-\ 
common IO controls for GR.303 devices.
.HP
.BR idlc (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR ISDN (4)\ \-\ 
OpenSS7 ISDN Devices.
.HP
.BR isdn (4)\ \-\ 
Integrated Services Digital Network (ISDN) Driver.
.HP
.BR isdn_undoc (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR lapd (4)\ \-\ 
Link Access Procedure\(ewD-Channel (LAPD) Driver.
.HP
.BR lapf (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR q921 (4)\ \-\ 
Link Access Procedure\(ewD-Channel (LAPD) Driver.
.HP
.BR q922 (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR q931 (4)\ \-\ 
Integrated Services Digital Network (ISDN) Driver.
.HP
.BR v52 (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR v5 (4)\ \-\ 
undocumented ISDN modules and devices.
.HP
.BR v5_ioctl (4)\ \-\ 
common IO controls for V5.2 devices.
.PD

.HP
.BR capi (7)\ \-\ 
Linux CAPI 2.0 Interface.
.PD 0
.HP
.BR cci (7)\ \-\ 
Call Control Interface.
.HP
.BR cci_capi (7)\ \-\ 
Call Control Interface\(emCorrigendum for Q.931/Q.932 Conformance
.HP
.BR cci_dass (7)\ \-\ 
Call Control Interface\(emCorrigendum for Q.931/Q.932 Conformance
.HP
.BR cci_gr303 (7)\ \-\ 
Call Control Interface\(emCorrigendum for Q.931/Q.932 Conformance
.HP
.BR cci_idlc (7)\ \-\ 
Call Control Interface\(emCorrigendum for Q.931/Q.932 Conformance
.HP
.BR cci_isdn (7)\ \-\ 
Call Control Interface\(emCorrigendum for Q.931/Q.932 Conformance
.PD

.HP
.BR CC_ADDR_ACK (7)\ \-\ 
provide call control address information.
.PD 0
.HP
.BR CC_ADDR_REQ (7)\ \-\ 
request call control address information.
.HP
.BR CC_ALERTING_IND (7)\ \-\ 
called CCS User alerted indication.
.HP
.BR CC_ALERTING_REQ (7)\ \-\ 
called CCS user alerted request.
.HP
.BR CC_BIND_ACK (7)\ \-\ 
bind protocol address acknowledgement.
.HP
.BR CC_BIND_REQ (7)\ \-\ 
bind protocol address request.
.HP
.BR CC_BLOCKING_CON (7)\ \-\ 
circuit management blocking confirmation.
.HP
.BR CC_BLOCKING_IND (7)\ \-\ 
circuit management blocking indication.
.HP
.BR CC_BLOCKING_REQ (7)\ \-\ 
circuit management blocking request.
.HP
.BR CC_BLOCKING_RES (7)\ \-\ 
circuit management blocking response.
.HP
.BR CC_CALL_FAILURE_IND (7)\ \-\ 
call failure indication.
.HP
.BR CC_CALL_REATTEMPT_IND (7)\ \-\ 
call reattempt indication.
.HP
.BR CC_CONNECT_IND (7)\ \-\ 
connection indication.
.HP
.BR CC_CONNECT_REQ (7)\ \-\ 
connection request.
.HP
.BR CC_CONT_CHECK_IND (7)\ \-\ 
continuity check indication.
.HP
.BR CC_CONT_CHECK_REQ (7)\ \-\ 
continuity check request.
.HP
.BR CC_CONT_REPORT_IND (7)\ \-\ 
continuity report indication.
.HP
.BR CC_CONT_REPORT_REQ (7)\ \-\ 
continuity report request.
.HP
.BR CC_CONT_TEST_IND (7)\ \-\ 
continuity test indication.
.HP
.BR CC_CONT_TEST_REQ (7)\ \-\ 
continuity test request.
.HP
.BR CC_DATALINK_FAILURE_IND (7)\ \-\ 
datalink failure indication.
.HP
.BR CC_DISCONNECT_IND (7)\ \-\ 
disconnect indication.
.HP
.BR CC_DISCONNECT_REQ (7)\ \-\ 
disconnect request.
.HP
.BR CC_ERROR_ACK (7)\ \-\ 
error acknowledgement.
.HP
.BR CC_ERROR_IND (7)\ \-\ 
error indication.
.HP
.BR CC_FACILITY_IND (7)\ \-\ 
facility indication.
.HP
.BR CC_FACILITY_REQ (7)\ \-\ 
facility request.
.HP
.BR CC_FORWXFER_IND (7)\ \-\ 
forward transfer indication.
.HP
.BR CC_FORWXFER_REQ (7)\ \-\ 
forward transfer request.
.HP
.BR CC_HOLD_CON (7)\ \-\ 
hold confirm.
.HP
.BR CC_HOLD_IND (7)\ \-\ 
hold indication.
.HP
.BR CC_HOLD_REQ (7)\ \-\ 
hold request.
.HP
.BR CC_HOLD_RES (7)\ \-\ 
hold response.
.HP
.BR CC_IBI_IND (7)\ \-\ 
in-band information indication.
.HP
.BR CC_IBI_REQ (7)\ \-\ 
in-band information request.
.HP
.BR CC_INFO_ACK (7)\ \-\ 
information acknowledgement.
.HP
.BR CC_INFO_REQ (7)\ \-\ 
information request.
.HP
.BR CC_INFORMATION_IND (7)\ \-\ 
information indication.
.HP
.BR CC_INFORMATION_REQ (7)\ \-\ 
information request.
.HP
.BR CC_INFO_TIMEOUT_IND (7)\ \-\ 
information timeout indication.
.HP
.BR CC_MAINT_IND (7)\ \-\ 
maintenance indication.
.HP
.BR CC_MORE_INFO_IND (7)\ \-\ 
more information indication.
.HP
.BR CC_MORE_INFO_REQ (7)\ \-\ 
more information request.
.HP
.BR CC_NOTIFY_IND (7)\ \-\ 
call control notification indication.
.HP
.BR CC_NOTIFY_REQ (7)\ \-\ 
call control notification request.
.HP
.BR CC_OK_ACK (7)\ \-\ 
successful receipt acknowledgement.
.HP
.BR CC_OPTMGMT_ACK (7)\ \-\ 
options management acknowledgement.
.HP
.BR CC_OPTMGMT_REQ (7)\ \-\ 
options management request.
.HP
.BR CC_PROCEEDING_IND (7)\ \-\ 
call proceeding indication.
.HP
.BR CC_PROCEEDING_REQ (7)\ \-\ 
call proceeding request.
.HP
.BR CC_PROGRESS_IND (7)\ \-\ 
call progress indication.
.HP
.BR CC_QUERY_CON (7)\ \-\ 
circuit query confirmation.
.HP
.BR CC_PROGRESS_REQ (7)\ \-\ 
call progress request.
.HP
.BR CC_QUERY_IND (7)\ \-\ 
circuit query indication.
.HP
.BR CC_QUERY_REQ (7)\ \-\ 
circuit query request.
.HP
.BR CC_QUERY_RES (7)\ \-\ 
circuit query response.
.HP
.BR CC_REGISTER_IND (7)\ \-\ 
registration indication.
.HP
.BR CC_REGISTER_REQ (7)\ \-\ 

NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/util/init/strisdn.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strisdn:	start and stop strisdn modules
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strisdn
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		ISDN character devices are present in the /dev directory and \
#		that the STREAMS ISDN kernel modules are configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strisdn
# Required-Start: streams strcompat $network
# Required-Stop: streams strcompat $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS Binary Compatibility Modules
# License: GPL
# Description:	This STREAMS ISDN init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS ISDN character
#	devices are present in the /dev directory and that the STREAMS ISDN
#	kernel modules are configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strisdn'
config="/etc/default/$name"
desc="the STREAMS ISDN modules"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRISDN_MKNOD" ; then
    for STRISDN_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $STRISDN_MKNOD ] ; then
	    break
	else
	    STRISDN_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$STRISDN_PRELOAD"       ] || STRISDN_PRELOAD=""
[ -n "$STRISDN_DRIVERS"       ] || STRISDN_DRIVERS=""
[ -n "$STRISDN_MODULES"       ] || STRISDN_MODULES=""
[ -n "$STRISDN_MAKEDEVICES"   ] || STRISDN_MAKEDEVICES="yes"
[ -n "$STRISDN_REMOVEDEVICES" ] || STRISDN_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRISDN_MKNOD" ] && STRISDN_MAKEDEVICES="no"
[ -z "$STRISDN_MKNOD" ] && STRISDN_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRISDN_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRISDN_MKNOD" -a :"$STRISDN_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS ISDN devices: "
	$STRISDN_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRISDN_MKNOD" -a ":$STRISDN_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS devices: "
	$STRISDN_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRISDN_PRELOAD $STRISDN_DRIVERS $STRISDN_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.8  2008-10-21 22:42:17  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.7  2008-04-29 00:49:02  brian
# - updated headers for release
#
# Revision 0.9.2.6  2007/08/14 07:41:35  brian
# - GPLv3 header update
#
# Revision 0.9.2.5  2007/03/08 22:42:57  brian
# - correct redirect
#
# Revision 0.9.2.4  2007/03/08 22:12:50  brian
# - update headers
#
# Revision 0.9.2.3  2007/03/08 22:08:23  brian
# - sense of VERBOSE reversed
#
# Revision 0.9.2.2  2007/03/08 20:25:28  brian
# - ubuntu looks for lsb info in init scripts
#
# Revision 0.9.2.1  2006/10/16 10:49:12  brian
# - added new package files
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
